testkit/src/api.rs

```diff
@@ -280,4 +279,58 @@ impl TestKitApi {
             false,
         )
     }
+
+    /// Converts iron Response to ApiError.
+    ///
+    /// # Panics
+    ///
+    /// - Panics if the response has a non-error response status.
+    fn response_to_api_error(response: Response) -> ApiError {
+        let status = response.status.expect("Status header is not set");
+
+        fn error(response: Response) -> String {
+            let body = response::extract_body_to_string(response);
+            serde_json::from_str::<JsonValue>(&body)
```

Comment:
I know this is probably my extra exposure to JS, but could it make sense to use [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression) here? (Or, to be less extravagant, another function.)

```rust
let body = response::extract_body_to_string(response);
(|| match serde_json::from_str::<JsonValue>(&body).ok()? {
    JsonValue::Object(ref object) if object.contains_key("description") => {
        Some(object["description"].as_str()?.to_owned())
    }
    JsonValue::String(string) => Some(string),
    _ => None,
})().unwrap_or(body)
```

Also note that I got rid of 2-way conversion `body -> value -> _: String` in the final match hand and changed the `_or_default()` logic - I don't see a clear case for losing the response body in some cases.

Reply:
Looks pretty laconically indeed! I like this approach, as well as IIFE )

Reply:
But `clippy` doesn't :(
(speaking of IIFE)